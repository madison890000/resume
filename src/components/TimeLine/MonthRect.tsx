import React from 'react';

const MonthRect = ({
                       gap = 2,
                       color,
                       index,
                       width = 20,
                   }: {
    index: number;
    width?: number
    color?: string;
    gap?: number;
}) => {
    return (
        <rect width={width} height="10" x={index * (width + gap)} y="5"
              fill={color}></rect>
    )
}

export default MonthRect
