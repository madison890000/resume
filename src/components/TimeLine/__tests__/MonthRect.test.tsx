import React from 'react';
import MonthRect from '../MonthRect';
import renderer from 'react-test-renderer';

test('renders MonthRect index 1', () => {
    const MonthRectDom = renderer.create(
        <MonthRect
            gap={2}
            color="red"
            index={1}
            width={20}
        />
    )
        .toJSON();
    expect(MonthRectDom)
        .toMatchSnapshot();
});

test('renders MonthRect index 2', () => {
    const MonthRectDom = renderer.create(
        <MonthRect
            gap={2}
            color="red"
            index={2}
            width={20}
        />
    )
        .toJSON();
    expect(MonthRectDom)
        .toMatchSnapshot();
});

test('renders MonthRect gap 4', () => {
    const MonthRectDom = renderer.create(
        <MonthRect
            gap={4}
            color="red"
            index={2}
            width={20}
        />
    )
        .toJSON();
    expect(MonthRectDom)
        .toMatchSnapshot();
});
test('renders MonthRect width 30', () => {
    const MonthRectDom = renderer.create(
        <MonthRect
            gap={4}
            color="red"
            index={2}
            width={30}
        />
    )
        .toJSON();
    expect(MonthRectDom)
        .toMatchSnapshot();
});

test('renders MonthRect color green', () => {
    const MonthRectDom = renderer.create(
        <MonthRect
            gap={4}
            color="green"
            index={2}
            width={30}
        />
    )
        .toJSON();
    expect(MonthRectDom)
        .toMatchSnapshot();
});
