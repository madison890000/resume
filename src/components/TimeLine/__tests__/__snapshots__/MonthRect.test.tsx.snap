// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MonthRect color green 1`] = `
<rect
  fill="green"
  height="10"
  width={30}
  x={68}
  y="5"
/>
`;

exports[`renders MonthRect gap 4 1`] = `
<rect
  fill="red"
  height="10"
  width={20}
  x={48}
  y="5"
/>
`;

exports[`renders MonthRect index 1 1`] = `
<rect
  fill="red"
  height="10"
  width={20}
  x={22}
  y="5"
/>
`;

exports[`renders MonthRect index 2 1`] = `
<rect
  fill="red"
  height="10"
  width={20}
  x={44}
  y="5"
/>
`;

exports[`renders MonthRect width 30 1`] = `
<rect
  fill="red"
  height="10"
  width={30}
  x={68}
  y="5"
/>
`;
